var/spack/repos/builtin/packages/msvc/package.py

```diff
@@ -57,6 +57,10 @@ def install(self, spec, prefix):
 
     @classmethod
     def determine_version(cls, exe):
+        if not exe.endswith(".exe") and not exe.endswith(".bat"):
+            # Not on windows, can't possibly be msvc
+            return
```

Comment:
If that's the case, shouldn't we make the code here tighter instead:
https://github.com/spack/spack/blob/52a995a95c372dfc50f30872129a3bfd91714753/lib/spack/spack/build_systems/compiler.py#L75-L77

?